lang/syn/src/parser/context.rs

```diff
@@ -181,6 +228,21 @@ impl ParsedModule {
         })
     }
 
+    fn unsafe_struct_fields(&self) -> impl Iterator<Item = &syn::Field> {
+        self.structs()
```

Comment:
One question is exactly what we want /// CHECKED: to mean.

If it is that both the owner and the type are verified, then skipping seeds and init isn't necessarily safe. Because one can use these keywords with UncheckedAccount or AccountInfo and subsequently forget the check (or in the case of init, fail to write the header and discriminator). For seeds, it may be the correct address, but the type might be data dependent. Obviously that's a bad design, but if one chooses to do such a thing, we can require them to use a comment to be explicit.

In either case, I think we want the CHECKED: comment to have a clear meaning w.r.t. the compile time guarantees and not be special cased anywhere.

Reply:
Yea, I think that is correct, it is easier to understand if its verification of the type as well. I always forget about the type. I'll remove the special casing here.

Reply:
Why is an account with constraints considered as an unsafe/unchecked account?
for eg:

```rs
#[account(
        address = global_market_state.base_token_mint @ ErrorCode::InvalidMintAccount
    )]
    pub base_mint_account: AccountInfo<'info>
```

Reply:
@SushantChandla because the `type` of `base_mint_account` is not explicitly asserted. if I understand the documentation correctly.